src/app/static/src/tests/components/projects/projectHistory.test.ts

```diff
@@ -463,6 +463,32 @@ describe("Project history component", () => {
         }
     });
 
+    it("can use carriage return to invokes promoteVersion action when confirm copy", async () => {
+        if (switches.promoteProject && !switches.renameProject) {
```

Comment:
Not sure what `!switches.renameProject` is doing here? Those switches are really obsolete now so we shouldn't use them  in new tests

Reply:
Me too, but without !switches.renameProject the test fails. This is the pattern found in other tests relating to this control.

Reply:
The switches are completely obsolete as both promoteProject and renameProject are completely implemented and always included in all deployments of HINT. If this test only passes with that conditional, it's because the conditional is always false since both those switches are in fact true, so the code inside the braces is never run. So there must be some other issue with the code I'm afraid. This pattern wouldn't have been a valid use of the switches anyway - it would only have been useful to test a situation where behaviour was particularly expected when promoteProject was true but renameProject was false. 

We should remove all references to those switches. I'll add a ticket to do that.